### PR TITLE
move topk to use OpInfo

### PR DESCRIPTION
**Pull request by albanD**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55547 move topk to use OpInfo**
* #55535 move logaddexp{2} to opinfo
* #55518 port addcdiv to opinfo
* #55517 port addcmul to OpInfo

Differential Revision: [D27649412](https://our.internmc.facebook.com/intern/diff/D27649412)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3318,6 +3342,14 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_out=False,
            sample_inputs_func=sample_inputs_max_min_reduction_no_dim,),
+    OpInfo('topk',
+           dtypes=all_types(),
+           dtypesIfCUDA=all_types_and(torch.bfloat16, torch.float16),
+           sample_inputs_func=sample_inputs_topk,
+           skips=(
+               # Topk is not raising a warning when the out is resized
+               SkipInfo('TestCommon', 'test_out'),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3351]: Nice skip + comment

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1078,6 +1078,30 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
...
     return inputs
 
+def sample_inputs_topk(op_info, device, dtype, requires_grad, **kwargs):
+    def get_tensor_input(size):
+        return make_tensor(size, device, dtype, requires_grad=requires_grad)
+
+    inputs = []
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3,)))
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3, 1)))
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3, -2)))
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1088]: Nice new case

**Review by mruberry — APPROVED** [review]
This is great! And thank you for adding some new cases, too!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (albanD) [reply torch/testing/_internal/common_methods_invocations.py:1088]: These are not new.
The old spec was specifying which args were dimension arguments and were automatically testing `val` and `val - size()` to make sure we do the wrapping properly.
I didn't find an arg in OpInfo to specify that so I added the tests manually.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:1110]: Cool!

**Pull request closed without merging**